### DISCUSSION TITLE
CMS Strapi  example - Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client

### DISCUSSION THREAD

**Discussion by msreekm** [vercel/next.js#14684] [Help]
# Bug report

## Describe the bug

CMS Strapi  example - Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client on the server console and broken SSR for this page

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. run cms-strapi-app example project , 
2. go to posts detail page (http://localhost:3001/posts/[slug])and do a full page refresh 
3. check the server console for errors
4. Not sure if it is the side effect of this error- SSR of the posts page is also broken (if you do a full page refresh, it shows the loading state first and then post details) . I noticed this in other CMS examples too ,so this may be a different issue.

`Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client at ServerResponse.setHeader (_http_outgoing.js:526:11) at DevServer.sendHTML (/Users/sreekumarmenon/projects/cms/cms-strapi-app/node_modules/next/dist/server/next-dev-server.js:23:5) at DevServer.render (/Users/sreekumarmenon/projects/cms/cms-strapi-app/node_modules/next/dist/next-server/server/next-server.js:49:37) at runMicrotasks (<anonymous>) at processTicksAndRejections (internal/process/task_queues.js:97:5) at async Object.fn (/Users/sreekumarmenon/projects/cms/cms-strapi-app/node_modules/next/dist/next-server/server/next-server.js:35:852) at async Router.execute (/Users/sreekumarmenon/projects/cms/cms-strapi-app/node_modules/next/dist/next-server/server/router.js:28:28) at async DevServer.run (/Users/sreekumarmenon/projects/cms/cms-strapi-app/node_modules/next/dist/next-server/server/next-server.js:44:494) at async DevServer.handleRequest (/Users/sreekumarmenon/projects/cms/cms-strapi-app/node_modules/next/dist/next-server/server/next-server.js:13:133) { code: 'ERR_HTTP_HEADERS_SENT' }`

## Expected behavior

No Error and Posts detail page is server rendered.

## System information

- OS: macOS
- Browser (if applies) All
- Version of Next.js: latest
- Version of Node.js: 12.16.1

**Comment by timneutkens** [comment]
Try upgrading to next@canary

**Reply by SalahAdDin** [reply]
@timneutkens After doing this i don't get that error, yes, but i still don't get any result.

**Comment by msreekm** [comment]
@timneutkens  thanks , http header error is fixed with v9.4.5-canary.23 , SSR is still broken (styling) ,it could be unrelated to the http header error, since it is happening for all CMS examples
